rfcs/0007-custom-event-metadata-from-annotations/README.md

```diff
@@ -0,0 +1,108 @@
+# RFC-0007 Custom Event Metadata from Annotations
+
+**Status:** provisional
+
+<!--
+Status represents the current state of the RFC.
+Must be one of `provisional`, `implementable`, `implemented`, `deferred`, `rejected`, `withdrawn`, or `replaced`.
+-->
+
+**Creation date:** 2024-05-23
+
+**Last update:** 2024-05-23
+
+## Summary
+
+Flux users often run into situations where they wish to send custom, static metadata fields defined
+in Flux objects on the events dispatched by the respective controller to notification-controller.
+This proposal offers a solution for supporting those use cases uniformly across all Flux controllers
+by sending annotation keys that are prefixed with a well-defined API Group.
+
+## Motivation
+
+The need for sending static metadata in Flux objects events can benefit users in a few ways:
+
+* When sending events about identical objects living in different clusters to the same notification channel,
+users often wish to differentiate those objects by the cluster where they live in. Sending custom metadata
+fields like `cluster`, `region` and `availabilityZone` would solve this problem.
+* When implementing integrations with notification-controller, users often wish to embed pieces of information
+in Flux objects that would help them identify entities or resources that are part of their integrations, e.g.
+the ID of a Deployment in the GitHub API.
+
+### Goals
+
+Provide a method for Flux users to embed custom/static fields in their Flux objects
+and have those fields sent to notification-controller as metadata on the event payload.
+
+### Non-Goals
+
+In this proposal we **do not** aim to provide a method for Flux users to send etcd-indexed custom metadata
+fields from Flux objects in events to notification-controller, most specifically labels. By design an event
+already contains enough identification information to locate the associated Flux object inside the cluster,
+which covers the use case of labels. Flux does not wish to incentive practices that are impactful to clusters
+without a strong reason or benefit.
+
+## Proposal
+
+When sending events about Flux objects, we propose sending annotation keys prefixed with a well-defined
+API Group in addition to those already sent by the controller to implement Flux-specific functionality.
+
+### User Stories
+
+* https://github.com/fluxcd/helm-controller/issues/680
+* https://github.com/fluxcd/helm-controller/pull/682#issuecomment-2079547110
```

Comment:
> It would be better to have verbally described user stories here for the scenarios in the issues and comments. Please see other RFCs for examples and the format of the user stories.

fixed, thanks!

@moritzschmitz-oviva thanks for the example!